diesel_migrations/migrations_internals/src/migration.rs

```diff
@@ -41,6 +32,12 @@ impl<'a> fmt::Display for MigrationName<'a> {
 }
 
 pub fn migration_from(path: PathBuf) -> Result<Box<Migration>, MigrationError> {
+    #[cfg(feature = "barrel")]
+    match ::barrel::diesel::migration_from(&path) {
```

Comment:
What do you think about changing this to `if let Some(migration) = ::barrel::diesel::migration_from(path)`?

Reply:
That would be more elegant although rustc tells me `attributes are not yet allowed on 'if' expressions`. The stabilisation is being tracked here: [rust-lang/rust#15701](https://github.com/rust-lang/rust/issues/15701). 

I'd suggest we change the syntax when the feature becomes stable, as it is more elegant.